### PR TITLE
Fix Into trait docs links 

### PR DESCRIPTION
**Pull request by 95th**
https://doc.rust-lang.org/std/convert/trait.Into.html

### CODE REVIEW

File: src/libcore/convert.rs [hunk 1]
```diff
@@ -213,7 +213,7 @@ pub trait AsMut<T: ?Sized> {
 /// # Generic Implementations
 ///
 /// - [`From`]`<T> for U` implies `Into<U> for T`
-/// - `Into` is reflexive, which means that `Into<T> for T` is implemented
+/// - [`Into`] is reflexive, which means that `Into<T> for T` is implemented
```
Comment (lzutao) [inline src/libcore/convert.rs]: There are no links to [`Into`]. You could write one.

Also the link in line 276 is unused:
https://github.com/rust-lang/rust/blob/a6a8f6c5b37597bf69178ee5cc6c185b7137f5d8/src/libcore/convert.rs#L276-L282

Comment (95th) [reply src/libcore/convert.rs]: Thanks. PTAL

File: src/libcore/convert.rs [hunk 2]
```diff
@@ -202,7 +202,7 @@ pub trait AsMut<T: ?Sized> {
 /// A value-to-value conversion that consumes the input value. The
 /// opposite of [`From`].
 ///
-/// One should only implement `Into` if a conversion to a type outside the current crate is
+/// One should only implement [`Into`] if a conversion to a type outside the current crate is
 /// required. Otherwise one should always prefer implementing [`From`] over `Into` because
```
Comment (GuillaumeGomez) [inline src/libcore/convert.rs]: You forgot one `Into`. ;)

File: src/libcore/convert.rs [hunk 3]
```diff
@@ -243,11 +243,11 @@ pub trait AsMut<T: ?Sized> {
 /// ```
 ///
 /// It is important to understand that `Into` does not provide a [`From`] implementation
```
Comment (GuillaumeGomez) [inline src/libcore/convert.rs]: You forgot another `Into`.

**Review by GuillaumeGomez — CHANGES_REQUESTED** [review]
You forgot a few `Into` (two other at line 254 and 257)

**Comment by 95th** [comment]
@GuillaumeGomez Do I need to create a separate link for `Into<T>` or the one for `Into` will work?

File: src/libcore/convert.rs (hunk 2 above)
Comment (95th) [reply src/libcore/convert.rs]: Done

File: src/libcore/convert.rs (hunk 3 above)
Comment (95th) [reply src/libcore/convert.rs]: Done

File: src/libcore/convert.rs [hunk 4]
```diff
@@ -242,19 +242,19 @@ pub trait AsMut<T: ?Sized> {
...
 ///
-/// Prefer using `Into` over [`From`] when specifying trait bounds on a generic function
-/// to ensure that types that only implement `Into` can be used as well.
+/// Prefer using [`Into`] over [`From`] when specifying trait bounds on a generic function
+/// to ensure that types that only implement [`Into`] can be used as well.
 ///
 /// # Examples
 ///
-/// [`String`] implements `Into<Vec<u8>>`:
+/// [`String`] implements [`Into<Vec<u8>>`]:
```
Comment (lzutao) [inline src/libcore/convert.rs]: I don't think we need to make a link here.

Comment (95th) [reply src/libcore/convert.rs]: @GuillaumeGomez Please let me know if this is ok

Comment (95th) [reply src/libcore/convert.rs]: I've removed the link from this one and update the one on Line 257.

**Comment by lzutao** [comment]
Could you squash all commits to one with message: `Fix Into trait links`?
Also change the PR title appropriately?

**Comment by 95th** [comment]
@lzutao I've made the suggested changes

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors: r+ rollup

**Review by GuillaumeGomez — APPROVED** [review]

**Pull request merged**